src/PrestaShopBundle/Utils/ImmutableFloat.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace PrestaShopBundle\Utils;
+
+/**
+ * Immutable value-object that contains a float value.
+ */
+class ImmutableFloat
```

Comment:
a "real VO" has to be in a VO namespace? O_o Evans knows this?

The conflict between float and Float is not IMHO a reason to add Immutable .. (moreover, core classes such as \DateTime have Immutable as suffix .. )

Moreover you are putting something resembling a Type, a VO, a Class into the "Utils" namespace .. I expect a util class will perform actions (and your class target seems to do, as it want to perform string parsing) .. in conclusion IMVHO there are naming issues in your PR, and names are important to make the code clear, understandable and less error prone.

It's my own opinion of course